### ISSUE TITLE
Game crashes when hosting local multiplayer game and specific client attempts to connect

### ISSUE THREAD

**Issue by sgerace** [OpenRA/OpenRA#19457]
## System Information

- Operating System:  macOS Mojave 10.14.6
- OpenRA Version: OpenRA engine version release-20210321
- OpenRA Mod: Red Alert mod version release-20210321
- Source: Official download package

## Exception log

```
OpenRA engine version release-20210321
Red Alert mod version release-20210321
Date: 2021-06-20 20:56:30Z
Operating System: OSX (Unix 18.7.0.0)
Runtime Version: Mono 6.4.0.198 (2019-06/fe64a4765e6 Wed Sep 18 14:14:51 EDT 2019) CLR 4.0.30319.42000
Exception of type `System.Net.Sockets.SocketException`: No route to host
  at System.Net.Sockets.Socket.SendTo (System.Byte[] buffer, System.Int32 offset, System.Int32 size, System.Net.Sockets.SocketFlags socketFlags, System.Net.EndPoint remoteEP) [0x00082] in <6aa0b19d109a447c94d1b43ec4471dc6>:0 
  at System.Net.Sockets.UdpClient.Send (System.Byte[] dgram, System.Int32 bytes, System.Net.IPEndPoint endPoint) [0x00067] in <6aa0b19d109a447c94d1b43ec4471dc6>:0 
  at BeaconLib.Beacon.ProbeReceived (System.IAsyncResult ar) [0x0006d] in <a0ceb97a865f4456b308a49b00346718>:0 
  at System.Net.Sockets.SocketAsyncResult+<>c.<Complete>b__27_0 (System.Object state) [0x0000b] in <6aa0b19d109a447c94d1b43ec4471dc6>:0 
  at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem () [0x00008] in <b814b509d4ad406fb40c6c93e38929e7>:0 
  at System.Threading.ThreadPoolWorkQueue.Dispatch () [0x00074] in <b814b509d4ad406fb40c6c93e38929e7>:0 
  at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback () [0x00000] in <b814b509d4ad406fb40c6c93e38929e7>:0 
```

## Additional information

I'm seeing this crash when attempting to host a game on a local network. Specifically, the game seems to crash when the second machine attempts to connect to the game hosted by the first machine (i.e., it crashes in the game setup screen). In this case, the first machine (the one hosting the game) crashes with the above exception log. Interestingly, if the second machine hosts the game, the first machine is unable to see the game. Additionally, we have a third machine and if that machine hosts the game, everything works fine (i.e., both the first and second machine are able to connect and we are able to play just fine). Additionally, the third machine is able to connect to the first machine without any issue.

So, in summary, it seems to just be the connection between the two specific machines. I've tried restarting both machines and disconnecting/reconnecting WiFi with no luck. We were able to play fine between these two machines fairly recently (within the last few weeks).

Any ideas of what might be causing this? It feels like it must be some sort of network configuration error, but I'm not sure what else to try.

**Comment by abcdefg30** [comment]
`No route to host` does sound like a network misconfiguration. (No default route?) I fear there is nothing we can do on our end.

**Comment by anvilvapre** [comment]
Make sure that you can ping all clients from the server host, I.e. `ping hostnameOrIp` from a terminal. Note that it does not always mean the game can connect after to.

The `SendTo`  suggests `UDP` protocol data is sent, rather than `TCP`. So perhaps a firewall on the server or client blocks `UDP` traffic. So likely on this single host a firewall is enabled - disallowing outgoing and/or incoming `UDP` traffic. I suspect the last - if the server does notice the client joining - but not being able to send something back.

The call originates from a 'beaconlib' used by the game. https://github.com/rix0rrr/beacon

**Comment by abcdefg30** [comment]
Closing due to inactivity and no clear path forward.
